### PR TITLE
Allow for empty tests list with plugin runner and running the same test repeatedly

### PR DESCRIPTION
**Pull request by alanking**
Still trying this out, but this will allow for plugin test hooks to fall back to the "default" behavior should no `--tests` option be used with the `run_plugin_tests.py` script.

### CODE REVIEW

File: irods_testing_environment/test_runner.py [hunk 1]
```diff
@@ -85,16 +87,19 @@ def result_string(self):
         r = '-----\nresults for [{}]\n'.format(self.name())
 
         r = r + '\tpassed tests:\n'
-        for test, duration in self.passed_tests().items():
-            r = r + '\t\t[[{:>9.4f}]s]\t[{}]\n'.format(duration, test)
+        for test, duration in self.passed_tests():
+            # TODO: a test list of type None may not indicate that all tests ran
+            r = r + '\t\t[[{:>9.4f}]s]\t[{}]\n'.format(duration, test or 'all tests')
```
Comment (korydraughn) [inline irods_testing_environment/test_runner.py:92]: What causes `'all tests'` to be printed?

**Review by korydraughn — COMMENTED** [review]
Nothing looked weird to me. Just had the one question.

File: irods_testing_environment/test_runner.py (hunk 1 above)
Comment (alanking) [reply irods_testing_environment/test_runner.py:92]: `test or 'all tests'` is equivalent to `test if test else 'all tests'` where `test == None` evaluates to `False`.

Comment (korydraughn) [reply irods_testing_environment/test_runner.py:92]: It was about L92. I was trying to understand what triggers the `'all tests'` text to be printed.

Additional comments are always good too :)

Comment (alanking) [reply irods_testing_environment/test_runner.py:92]: Okay, I will add a blurb to each test runner to describe the None case.

Comment (alanking) [reply irods_testing_environment/test_runner.py:92]: Done

File: irods_testing_environment/test_manager.py [hunk 2]
```diff
@@ -11,6 +11,14 @@ class test_manager:
     def __init__(self, containers, tests, test_type='irods_python_suite'):
         """Constructor for `test_manager`.
 
+        A note about passing None to `tests`:
+        None is a special value which indicates that no test list was specified. The test
```
Comment (korydraughn) [inline irods_testing_environment/test_manager.py:15]: I noticed that all references to **None** are surrounded by backticks except the one on this line.

**Review by korydraughn — COMMENTED** [review]
Just one more edit and I think we're good here.

File: irods_testing_environment/test_manager.py (hunk 2 above)
Comment (alanking) [reply irods_testing_environment/test_manager.py:15]: Done

Comment (korydraughn) [reply irods_testing_environment/test_manager.py:15]: I actually was referring to L15.

"None is a special ..."

Comment (alanking) [reply irods_testing_environment/test_manager.py]: Whoops. Now done.

**Review by korydraughn — APPROVED** [review]
Ship it!
